js/classes/entities/midground/weapon/projectile_base.js

```diff
@@ -62,11 +67,41 @@ class ProjectileBase {
 
         this.projectile.pos = Vector.add(this.projectile.pos, new Vector(pos.x, pos.y + this.projectile.yVelocity));
 
+        // update the rotation
+        let angle = Math.atan2(this.projectile.dir.y, this.projectile.dir.x);
+        if (angle < 0) {
+            angle += 2 * Math.PI; // convert to 0 to 2π range
+        }
+        this.projectile.rotation = angle;
+        
+        if (angle >= 0 && angle < Math.PI * 0.125 || angle >= Math.PI * 1.875) {
```

Comment:
Wow this is a lot. If it works that's great but I can't help but wonder how this could be cleaned up. Perhaps a function with parameters which would be the values that change in the conditions such as 0.125 etc and whatever the increase or decrease in the value could be added to the value depending on the direction. For example the delta in the values you change from right to rightDown to Down, are they all the same delta value? Perhaps a pattern arises and this could be made into a generic function.